### PR TITLE
Wallet button pref

### PR DESCRIPTION
**Pull request by bbondy**
Adds a "show wallet button" option to Wallet settings. It is on by default as long as the chrome://flags feature is enabled.

If the wallet button is hidden and the panel is show, the panel is anchored to the app menu (hamburger menu)

Resolves https://github.com/brave/brave-browser/issues/15550

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`chromium_src` change ++

**Review by emerick — APPROVED** [review]
LGTM

File: browser/ui/views/toolbar/wallet_button.h [hunk 1]
```diff
@@ -50,9 +51,11 @@ class WalletButton : public ToolbarButton, public views::WidgetObserver {
   // views::WidgetObserver:
   void OnWidgetDestroying(views::Widget* widget) override;
 
-  WebUIBubbleManagerT<WalletPanelUI> webui_bubble_manager_;
+  std::unique_ptr<WebUIBubbleManagerT<WalletPanelUI>> webui_bubble_manager_;
 
   PrefService* prefs_ = nullptr;
+  View* backup_anchor_view_ = nullptr;
+  Profile* profile_;
```
Comment (tmancey) [inline browser/ui/views/toolbar/wallet_button.h:58]: nit: `.. = nullptr;  // NOT OWNED'

**Review by tmancey — APPROVED** [review]
LGTM++

**Comment by bbondy** [comment]
Linux CI failure - Unrelated:
```
1 test crashed:

[2021-05-05T15:50:03.787Z]     RewardsPublisherBrowserTest.VisitVerifiedPublisher (../../brave/components/brave_rewards/browser/test/rewards_publisher_browsertest.cc:138)
```

File: browser/ui/views/toolbar/wallet_button.h (hunk 1 above)
Comment (bbondy) [reply browser/ui/views/toolbar/wallet_button.h:58]: will do in a follow up, thx.

**Pull request merged**